### PR TITLE
Added MachineTokenProvider interface with different implementations

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
Adds MachineTokenProvider interface. There are different implementations for single and multiuser packagings. With these changes, there is no necessity to override anything for implementations of infrastructures in `che multiuser` assembly.

It also contains minor cleaning up of jpa tests and OpenShift provisioning.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/6587

### CODE REVIEW

**Review by skabashnyuk — APPROVED** [review]

**Review by gazarenkov — APPROVED** [review]

**Review by akorneta — APPROVED** [review]

**Review by garagatyi — APPROVED** [review]

**Review by riuvshin — APPROVED** [review]

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/hc/ServersChecker.java [hunk 1]
```diff
@@ -56,16 +60,17 @@
...
       RuntimeIdentity runtimeIdentity,
       String machineName,
       Map<String, ? extends Server> servers,
-      ServerCheckerFactory serverCheckerFactory) {
+      @Assisted MachineTokenProvider machineTokenProvider) {
     this.runtimeIdentity = runtimeIdentity;
     this.machineName = machineName;
     this.servers = servers;
-    this.serverCheckerFactory = serverCheckerFactory;
     this.timer = new Timer("ServerReadinessChecker", true);
```
Comment (garagatyi) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/hc/ServersChecker.java]: Can you align string constant with the new new name

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/hc/ServersChecker.java [hunk 2]
```diff
@@ -56,16 +60,17 @@
...
    * @param servers map of servers in a machine
    */
-  public ServersReadinessChecker(
+  @Inject
+  public ServersChecker(
       RuntimeIdentity runtimeIdentity,
       String machineName,
       Map<String, ? extends Server> servers,
-      ServerCheckerFactory serverCheckerFactory) {
+      @Assisted MachineTokenProvider machineTokenProvider) {
```
Comment (garagatyi) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/hc/ServersChecker.java]: Am I wrong that assisted annotation should be on other arguments, but not on this one?

**Review by garagatyi — APPROVED** [review]
In overall looks good. I would try to avoid interceptor, but since it is not your code we can discuss it separately.
Please, make sure that it is tested on different infrastructures and deployments.

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/hc/ServersChecker.java (hunk 2 above)
Comment (sleshchenko) [reply wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/hc/ServersChecker.java]: You're correct. I've already fixed it locally.

**Pull request merged**